### PR TITLE
I2 fav index

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
Add favorites index page with tests, missing one test for favorites poro, unsure of how to implement.

### CODE REVIEW

File: app/poro/favorites.rb [hunk 1]
```diff
@@ -15,4 +15,16 @@ def add_pet(id)
...
     @contents[id.to_s].to_i
   end
+
+  def extract_ids
+    @ids = @contents.map do |key, value|
+      key.to_i
+    end
+  end
+
+  def translate_ids
```
Comment (PhilipDeFraties) [inline app/poro/favorites.rb:25]: This is the method I cant figure out a test for, so far with the favorites poro tests its been really simple example data to test such as 
`favorites = Favorites.new({
          '1' => 1,
          '2' => 1
        })`
However if I need to test its ability to find an actual pet based on an extracted id that would mean creating pets for the test, favoriting them, passing the favorited info into the method somehow... Idk I cant wrap my head around it.

Comment (RyanDBarnett) [reply app/poro/favorites.rb:25]: I'm not sure but we might be able to mock @contents with a simple hash like { '1' => 1, '2' => 1 } and then expect extract_ids to return [1, 2].

Comment (RyanDBarnett) [reply app/poro/favorites.rb:25]: I looked at what you were trying to do some more and I think I came up with a solution. I have created a i2-fav-index-2 PR with my suggested changes.

Comment (RyanDBarnett) [reply app/poro/favorites.rb:25]: I am going to merge this because it works.

**Pull request merged**